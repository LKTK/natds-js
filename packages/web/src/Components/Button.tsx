import React, { FunctionComponent } from 'react';
import { withTheme } from '@material-ui/styles';
import MaterialButton, { ButtonProps as IButtonProps } from '@material-ui/core/Button';
export { ButtonProps as IButtonProps } from '@material-ui/core/Button';

const Button: FunctionComponent<IButtonProps> = (props: IButtonProps) => {
  return <MaterialButton {...props} />;
};

export default withTheme(Button);
