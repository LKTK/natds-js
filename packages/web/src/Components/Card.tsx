import React, { FunctionComponent, forwardRef } from 'react';
import { withTheme } from '@material-ui/styles';
import MaterialCard, { CardProps as ICardProps } from '@material-ui/core/Card';
export { CardProps as ICardProps } from '@material-ui/core/Card';

export const Card: FunctionComponent<ICardProps> = forwardRef((
  props: ICardProps,
  ref: any
) => <MaterialCard {...props} ref={ref} />);

export default withTheme(Card);
