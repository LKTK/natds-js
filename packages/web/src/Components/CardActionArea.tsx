import React, { FunctionComponent, forwardRef } from "react";
import { withTheme } from "@material-ui/styles";
import MaterialCardActionArea, {
  CardActionAreaProps as ICardActionAreaProps,
} from "@material-ui/core/CardActionArea";

export { CardActionAreaProps as ICardActionAreaProps } from "@material-ui/core/CardActionArea";

const CardActionArea: FunctionComponent<ICardActionAreaProps> = forwardRef((
  props: ICardActionAreaProps,
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ref: any,
) => <MaterialCardActionArea {...props} ref={ref} />);

CardActionArea.displayName = "CardActionArea";

export default withTheme(CardActionArea);
