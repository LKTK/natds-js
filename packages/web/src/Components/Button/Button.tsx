import * as React from "react";

import MaterialButton from "@material-ui/core/Button";
import { IButtonProps } from "./Button.props";
import useStyles from "./Button.styles";

export { IButtonProps } from "./Button.props";

/**
 * ```
 * import { Button } from '@naturacosmeticos/natds-web';
 * ```
 *
 * For more advanced usages, check [Material UI Button docs](https://material-ui.com/components/buttons/)
 */
export const Button = React.forwardRef<unknown, IButtonProps>(
  (
    props: IButtonProps,
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    ref: any,
  ) => {

    const { classes, ...otherProps } = props;

    const mergedClasses = {
      ...classes,
      ...useStyles(),
    };

    return <MaterialButton
      classes={mergedClasses}
      ref={ref}
      {...otherProps}
    />;

  });

Button.displayName = "Button";
export default Button;
