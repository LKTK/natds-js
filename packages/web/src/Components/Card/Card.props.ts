import { CardProps } from "@material-ui/core/Card";

export interface ICardProps extends CardProps {

  /**
   * The content of the component.
   *
   * @optional
   * @type node
   */
  children?: CardProps["children"]

  /**
   * Override or extend the styles applied to the component.
   *
   * @optional
   * @type object
   */
  classes?: CardProps["classes"]

  /**
   * If `true`, the card will use raised styling.
   *
   * @default false
   * @optional
   * @type bool
   */
  raised?: CardProps["raised"]
}
