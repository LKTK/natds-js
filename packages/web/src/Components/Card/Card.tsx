import * as React from "react";
import MaterialCard from "@material-ui/core/Card";
import { ICardProps } from "./Card.props";

/**
 * For more advanced usages, check [Material Card docs](https://material-ui.com/components/cards/)
 *
 * ## Importing
 *
 * ```
 * import { Card } from "@naturacosmeticos/natds-web";
 * ```
 *
 * @see https://material-ui.com/components/cards/
 */
export const Card = React.forwardRef<unknown, ICardProps>(
  (props: ICardProps, ref: ICardProps["ref"]) => (
    <MaterialCard {...props} ref={ref} />
  ),
);

Card.displayName = "Card";

export default Card;
