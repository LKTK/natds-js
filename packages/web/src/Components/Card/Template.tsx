import { Story } from "@storybook/react";
import * as React from "react";
import createStyles from "@material-ui/core/styles/createStyles";
import makeStyles from "@material-ui/core/styles/makeStyles";
import { ICardProps } from "./Card.props";
import { Card } from "./Card";
import { Collapse, ICollapseProps } from "../Collapse";
import { WithIconButtonsAndDisabledSpacing } from "../CardActions/CardActions.stories";
import { IconButton } from "../IconButton";
import { CardHeader } from "../CardHeader";
import { Playground as CardHeaderPlayground } from "../CardHeader/CardHeader.stories";
import { CardMedia } from "../CardMedia";
import { Playground as CardMediaPlayground } from "../CardMedia/CardMedia.stories";
import { CardContent } from "../CardContent";
import { WithBody2Typography } from "../CardContent/CardContent.stories";
import { CardActions } from "../CardActions";
import { ExpandExample } from "../IconButton/IconButton.stories";
import { ForCardContent } from "../Collapse/Collapse.stories";

export const Template : Story<ICardProps> = (args : ICardProps) => (
  <Card {...args} />
);

export const PlaygroundTemplate : Story<ICardProps> = (args: ICardProps) => {

  const [expanded, setExpanded] = React.useState<ICollapseProps["in"]>(false);

  const handleExpandClick = () => {
    setExpanded((oldValue: ICollapseProps["in"]) => !oldValue);
  };

  const [AddToFavorites, Share] = WithIconButtonsAndDisabledSpacing.args.children as Array<typeof IconButton>;

  /**
   * @todo use time/duration token from `natds-themes` ssot
   */
  const transition = "transform 0.1s ease-in-out";

  const style = createStyles({
    collapsedIcon: {
      transform: "rotate(0deg)",
      transition,
    },
    expandedIcon: {
      transform: "rotate(180deg)",
      transition,
    },
  });

  const useStyles = makeStyles(style, { name: "CardDemo" });

  const { collapsedIcon, expandedIcon } = useStyles();

  return (
    <Card {...args}>
      <CardHeader {...CardHeaderPlayground.args} />
      <CardMedia {...CardMediaPlayground.args} />
      <CardContent {...WithBody2Typography.args} />
      <CardActions {...WithIconButtonsAndDisabledSpacing.args}>
        {AddToFavorites}
        {Share}
        <IconButton {...ExpandExample.args}
          aria-expanded={expanded}
          className={expanded ? expandedIcon : collapsedIcon}
          onClick={handleExpandClick} />
      </CardActions>
      <Collapse {...ForCardContent.args}
        in={expanded} />
    </Card>
  );
};

export default Template;
