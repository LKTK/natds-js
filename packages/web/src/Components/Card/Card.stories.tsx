import { Meta, Story } from "@storybook/react";
import * as React from "react";
import { Card } from "./Card";
import { CardActions } from "../CardActions";
import { CardContent } from "../CardContent";
import { CardHeader } from "../CardHeader";
import { CardMedia } from "../CardMedia";
import { ICardProps } from "./Card.props";
import { WithHeading6AndBody2, WithMultipleTypographies } from "../CardContent/CardContent.stories";
import { PlaygroundTemplate, Template } from "./Template";
import { Playground as CardActionsPlayground } from "../CardActions/CardActions.stories";
import { Playground as CardMediaPlayground } from "../CardMedia/CardMedia.stories";

export default {
  component: Card,
  subcomponents: {
    CardActions, CardContent, CardHeader, CardMedia,
  },
  title: "Components/Card",
} as Meta;

export const Playground : Story<ICardProps> = PlaygroundTemplate.bind({});
Playground.args = {
  style: { maxWidth: 328 },
};

export const Simple : Story<ICardProps> = Template.bind({});
Simple.args = {
  children: [
    <CardContent key={"content"} {...WithMultipleTypographies.args} />,
    <CardActions key={"actions"} {...CardActionsPlayground.args} />,
  ],
  style: { maxWidth: 328 },
};

export const WithMedia : Story<ICardProps> = Template.bind({});
WithMedia.args = {
  children: [
    <CardMedia key={"media"} {...CardMediaPlayground.args} />,
    <CardContent key={"content"} {...WithHeading6AndBody2.args} />,
    <CardActions key={"actions"} {...CardActionsPlayground.args} />,
  ],
  style: {
    maxWidth: 328,
  },
};
