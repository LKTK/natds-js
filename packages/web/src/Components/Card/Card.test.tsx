import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { Card } from "./Card";
import { CardActionArea } from "../CardActionArea";
import { CardActions } from "../CardActions";
import { Button } from "../Button";
import { CardContent } from "../CardContent";
import { Typography } from "../Typography";
import { CardHeader } from "../CardHeader";
import { CardMedia } from "../CardMedia";
import { Collapse } from "../Collapse";
import { Paper } from "../Paper";

describe("Card components", () => {
  describe("rendering variants", () => {
    test("should match to snapshot - empty card", () => {
      const testRenderer = TestRenderer.create(<Card />);

      expect(testRenderer).toMatchSnapshot("Card empty snapshot");
    });

    test("should match to snapshot - with action area", () => {
      const testRenderer = TestRenderer.create(<Card>
        <CardActionArea>Card action area test text</CardActionArea>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with action area snapshot");
    });

    test("should match to snapshot - with actions", () => {
      const testRenderer = TestRenderer.create(<Card>
        <CardActions>
          <Button>Button action text</Button>
        </CardActions>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with actions snapshot");
    });

    test("should match to snapshot - with content", () => {
      const testRenderer = TestRenderer.create(<Card>
        <CardContent>
          <Typography>Content text</Typography>
        </CardContent>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with content snapshot");
    });

    test("should match to snapshot - with header", () => {
      const testRenderer = TestRenderer.create(<Card>
        <CardHeader title="Header 1" subheader="Header 2" />
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with header snapshot");
    });

    test("should match to snapshot - with media", () => {
      const testRenderer = TestRenderer.create(<Card>
        <CardMedia title="media image">Card media image placeholder</CardMedia>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with media snapshot");
    });

    test("should match to snapshot - with collapse", () => {
      const testRenderer = TestRenderer.create(<Card>
        <Collapse in={true} timeout="auto" unmountOnExit>
          <Typography>Collapse text</Typography>
        </Collapse>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with collapse snapshot");
    });

    test("should match to snapshot - with paper", () => {
      const testRenderer = TestRenderer.create(<Card>
        <Paper>
          <Typography>Paper text</Typography>
        </Paper>
      </Card>);

      expect(testRenderer).toMatchSnapshot("Card with paper snapshot");
    });
  });
});
