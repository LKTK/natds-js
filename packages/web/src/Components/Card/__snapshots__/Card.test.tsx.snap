// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components rendering variants should match to snapshot - empty card: Card empty snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
/>
`;

exports[`Card components rendering variants should match to snapshot - with action area: Card with action area snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    Card action area test text
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with actions: Card with actions snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardActions-root MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root NatDSButton-root-1 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label NatDSButton-label-2"
      >
        Button action text
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with collapse: Card with collapse snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          Collapse text
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with content: Card with content snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Content text
    </p>
  </div>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with header: Card with header snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardHeader-root"
  >
    <div
      className="MuiCardHeader-content"
    >
      <span
        className="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
      >
        Header 1
      </span>
      <span
        className="MuiTypography-root MuiCardHeader-subheader MuiTypography-body1 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
      >
        Header 2
      </span>
    </div>
  </div>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with media: Card with media snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root"
    title="media image"
  >
    Card media image placeholder
  </div>
</div>
`;

exports[`Card components rendering variants should match to snapshot - with paper: Card with paper snapshot 1`] = `
<div
  className="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      Paper text
    </p>
  </div>
</div>
`;
